apps/src/applab/ViewDataButton.jsx

```diff
@@ -0,0 +1,49 @@
+/** @file Button that opens the data browser */
+// Strict linting: Absorb into global config when possible
+/* jshint
+ unused: true,
+ eqeqeq: true,
+ maxlen: 120
+ */
+/* global $ */
+
+var applabMsg = require('./locale');
+var styles = require('./PlaySpaceHeaderStyles');
+
+/**
+ * The button above the visualization that opens the data browser.
+ * @type {function}
+ */
+var ViewDataButton = React.createClass({
+  propTypes: {
+    onClick: React.PropTypes.func.isRequired
+  },
+
+  render: function () {
+
+    var showDataButtonStyle = $.extend(
+        {
+          float: 'right',
+          textAlign: 'left',
+          maxWidth: '100%',
+          whiteSpace: 'nowrap',
+          overflow: 'hidden',
+          textOverflow: 'ellipsis'
+        },
+        styles.buttonStyle,
+        styles.inactiveStyle
```

Comment:
In this case I'm just trying to preserve the pre-refactor behavior.  Agree that this is weird.  I'll change it up and try to verify it doesn't break anything.